src/Vector.cpp

```diff
@@ -0,0 +1,119 @@
+
+//
+// Created by Dvir Avraham on 19/03/2023.
+//
+
+#include "Vector.h"
+
+// implemnt vector
+
+// constructor with size and value
+Vector::Vector(size_t size, int value) {
+    _data = new int[size];
+    _size = size;
+    _capacity = size;
+    for (int i = 0; i < size; i++) {
+        _data[i] = value;
+    }
+}
+
+// destructor
+Vector::~Vector() {
+    delete[] _data;
+}
+
+//
+
+Vector::Vector(const Vector &other) {
+    _data = new int[other._size];
```

Comment:
notice that you might be allocating a 0 sized array int. This will essentially means that no memory is allocated and thus will require another allocation when an item is inserted (hurt performance). It is probably wiser to pre allocate some memory in the array